numpy/core/src/umath/ufunc_object.c

```diff
@@ -2219,25 +2264,51 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
     /* Get all the arguments */
     retval = get_ufunc_arguments(ufunc, args, kwds,
                 op, &order, &casting, &extobj,
-                &type_tup, &subok, NULL, &axes);
+                &type_tup, &subok, NULL, &axes, &keepdims);
     if (retval < 0) {
         goto fail;
     }
-
+    /*
+     * If keepdims was passed in (and thus changed from the initial value
+     * on top), check the gufunc is suitable.
+     */
+    if (keepdims != -1) {
+        retval = _check_keepdims_support(ufunc);
+        if (retval < 0) {
```

Comment:
nit: would probably combine these two lines - we do that pretty frequently

Reply:
As in `if (retval = ... < 0)`? I don't see any example of that in `ufunc_object`, and personally don't like it much. Anyway, will await for the more thorough look.

Reply:
I mean:
```
if (_check_keepdims_support(ufunc) < 0) {
    return -1;
}
```

Reply:
But `fail` needs to have `retval` set...

Reply:
What am I missing here? Is `retval` sometimes not `-1`?

Reply:
It left at `0` by the `get_ufunc_arguments` statement above. There is a debug statement in `fail` that prints `retval`, but to be honest I'm not sure it ever has a value other than -1 for failure. Probably would make sense to clean that all up, but perhaps in another PR...